### ISSUE TITLE
Push Style crash when Window Width is auto (0,0)

### ISSUE THREAD

**Issue by Strepto** [StereoKit/StereoKit#651]
## Description

Crash when PushTextStyle and then rendering a Text when a Window is using auto size (0,0).

Using  dotnet `StereoKit v0.3.7.preview.9`
(Slightly edited example as I figured out a better repro)
```cs
UI.WindowBegin(
    "Cool window",
    ref _infoWindowPose,
    new Vec2(0, 0), // Also crashes when I use the WindowBegin variant without the size param.
    UIWin.Normal,
    UIMove.FaceUser
);

var largeText2 = Text.MakeStyle(Font.Default, 0.04f, Color.White);
UI.PushTextStyle(largeText2);
UI.Text("Crash!"); // Crashes only when using non-default TextStyle
UI.PopTextStyle();
UI.WindowEnd();
```

```log
Fatal error. System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Repeat 2 times:
--------------------------------
   at StereoKit.NativeAPI.ui_text_16(System.String, StereoKit.TextAlign)
--------------------------------
   at Common.App.Step()
   at Program+<>c__DisplayClass0_0.<<Main>$>b__0()
   at StereoKit.SK._Step()
   at StereoKit.NativeAPI.sk_run(System.Action, System.Action)
   at StereoKit.NativeAPI.sk_run(System.Action, System.Action)
   at StereoKit.SK.Run(System.Action, System.Action)
   at Program.<Main>$(System.String[])
```

## Platform / Environment

I have only tested this in flatscreen with dotnet.

When NOT using UI.PushTextStyle this does not crash.

Not sure if this is a short self-contained correct example (SSCCE). As it seems like a bit of a strange issue I'm primarily posting this to understand if its a known issue before making a bigger repro.

## Workaround

Specify a X width of the WindowBegin: 
`UI.WindowBegin("title", ref pose, new Vec2(1,0))`

**Comment by maluoi** [comment]
I can confirm this crashes on preview.8! Thanks for an excellent repro sample :)

I think I saw it once on 9 as well, but I overwrote my local copy of 9 with more recent dev code at some point, and I can't decide if it's just caching that somewhere or actually using 9. However, latest on my computer doesn't cause issues. Preview.9 had a few layout issues to it, and I fixed a number of related problems. This was probably covered by one of those fixes.

I may not be able to publish preview.10 in the next week or two, but if you need an early build of that, send me a message on Discord :)

Just in case you're interested, I tested this in StereoKitTest like this to try out the latest code:
```csharp
using StereoKit;

internal class TestPushStyle : ITest
{
	Pose _infoWindowPose = new Pose(0, 0, -0.5f, Quat.LookDir(-Vec3.Forward));
	public void Step()
	{
		int SelectedTreeIndex = 1;
		Vec3? SelectionPosition = new Vec3(0, 0, -1);
		if (SelectedTreeIndex > 0 && SelectionPosition.HasValue)
		{
			Lines.AddAxis(_infoWindowPose, 1, 0.1f);
			UI.WindowBegin(
				"Part info for " + SelectedTreeIndex,
				ref _infoWindowPose,
				new Vec2(0, 0), // Auto width / height
				UIWin.Normal,
				UIMove.FaceUser
			);
			var largeText = Text.MakeStyle(Font.Default, 0.04f, Color.White);
			UI.PushTextStyle(largeText);
			UI.Text("TreeIndex: " + SelectedTreeIndex); // <- Crashes
			UI.PopTextStyle();
			UI.WindowEnd();
		}
	}

	public void Initialize() { }
	public void Shutdown() { }
}
```

And in a standalone project to check older packages:
```csharp
using StereoKit;

if (!SK.Initialize())
	return;

Pose _infoWindowPose = new Pose(0, 0, -0.5f, Quat.LookDir(-Vec3.Forward));
int   SelectedTreeIndex = 1;
Vec3? SelectionPosition = new Vec3(0, 0, -1);
SK.Run(() =>
{
	if (SelectedTreeIndex > 0 && SelectionPosition.HasValue)
	{
		Lines.AddAxis(_infoWindowPose, 1, 0.1f);
		UI.WindowBegin(
			"Part info for " + SelectedTreeIndex,
			ref _infoWindowPose,
			new Vec2(0, 0), // Auto width / height
			UIWin.Normal,
			UIMove.FaceUser
		);
		var largeText = Text.MakeStyle(Font.Default, 0.04f, Color.White);
		UI.PushTextStyle(largeText);
		UI.Text("TreeIndex: " + SelectedTreeIndex); // <- Crashes
		UI.PopTextStyle();
		UI.WindowEnd();
	}
});
```

**Comment by Strepto** [comment]
If its fixed in v.next its no rush! The workaround of forcing a width works fine for me 👍

Feel free to close whenever you feel like 🚀

**Comment by maluoi** [comment]
This should be fixed in 0.3.7-preview.1570!
